sklearn/cluster/hierarchical.py

```diff
@@ -704,9 +729,21 @@ def fit(self, X, y=None):
                              " instance, got 'memory={!r}' instead.".format(
                                  type(memory)))
 
-        if self.n_clusters <= 0:
+        # if distance threshold is set we set n_clusters to None
```

Comment:
How do I handle the case when both `n_clusters` and `distance_threshold` are set? With default `n_clusters=2`, when `distance_threshold` is set it fails.

I had tried to get around this earlier by changing the default `n_clusters` to `None`, but that fails other tests. This was a compromise, but I'm not sure.